dds/DCPS/WriteDataContainer.cpp

```diff
@@ -1500,6 +1488,159 @@ WriteDataContainer::log_send_state_lists (OPENDDS_STRING description)
              instances_.size()));
 }
 
+void
+WriteDataContainer::set_deadline_period(const TimeDuration& deadline_period)
+{
+  // Call comes from DataWriterImpl_t which should arleady have the lock_.
+
+  // Reset the deadline timer if the period has changed.
+  if (deadline_period_ != deadline_period) {
+    if (deadline_period_ == TimeDuration::max_value) {
+      OPENDDS_ASSERT(deadline_map_.empty());
+
+      const MonotonicTimePoint deadline = MonotonicTimePoint::now() + deadline_period;
+
+      for (PublicationInstanceMapType::iterator iter = instances_.begin();
+           iter != instances_.end();
+           ++iter) {
+        iter->second->deadline_ = deadline;
+        deadline_map_.insert(std::make_pair(deadline, iter->second));
+      }
+
+      if (!deadline_map_.empty()) {
+        deadline_task_->schedule(deadline_period);
+      }
+    } else if (deadline_period == TimeDuration::max_value) {
+      if (!deadline_map_.empty()) {
+        deadline_task_->cancel();
+      }
+
+      deadline_map_.clear();
+    } else {
+      // TODO: Which is correct?  The first can cause immediate deadline misses?  The second corresponds to the existing behavior.
+      //const TimeDuration adjustment = deadline_period - deadline_period_;
+      const TimeDuration adjustment = deadline_period;
+      DeadlineMapType new_map;
+      for (PublicationInstanceMapType::iterator iter = instances_.begin();
+           iter != instances_.end();
+           ++iter) {
+        iter->second->deadline_ += adjustment;
```

Comment:
So if the previous deadline_ was an hour in the future, but new deadline is 5 seconds, this looks like it's bumping the next deadline and next wait to 1hr + 5 seconds, which seems wrong?

Reply:
I agree.  I will change it so it resets all instances to now() + deadline_period.